docs/content/development/extensions-core/kafka-ingestion.md

```diff
@@ -222,6 +222,18 @@ GET /druid/indexer/v1/supervisor/<supervisorId>/history
 ```
 Returns an audit history of specs for the supervisor with the provided ID.
 
+#### Reset Supervisor
+```
+POST /druid/indexer/v1/supervisor/<supervisorId>/reset
+```
+The indexing service keeps track of the latest persisted Kafka offsets in order to provide exactly-once ingestion
+guarantees across tasks. Subsequent tasks must start reading from where the previous task completed in order for the
+generated segments to be accepted. If the messages at the expected starting offsets are no longer available in Kafka
+(typically because the message retention period has elapsed) the supervisor will refuse to start and in-flight tasks
+will fail. This endpoint can be used to clear the stored offsets which will cause the supervisor to start reading from
+either the earliest or latest offsets in Kafka (depending on the value of `useEarliestOffset`). **Use of this endpoint
+will result in data loss or data duplication.**
```

Comment:
@gianm made the doc changes but also made the implementation more aggressive to not only clear the metadata but also kill any active non-publishing task and re-create it with valid offsets.

I couldn't think of any cases where the user would want to clear the metadata but still keep the active tasks - I suppose in the case of a catastrophic Kafka failure they may not want to kill the active tasks since they may contain data no longer available in Kafka, but in that case they can shutdown the supervisor which will force the tasks to publish and then restart the supervisor and reset at that point.

Having it automatically kill the tasks should be a better user experience since they won't have to either manually kill tasks or shutdown/resubmit the supervisor if the tasks have been initialized with bad offsets.